storm-core/src/clj/org/apache/storm/daemon/supervisor.clj

```diff
@@ -301,7 +303,9 @@
         (try
           (rmpath (ConfigUtils/workerPidPath conf id pid))
           (catch Exception e)))) ;; on windows, the supervisor may still holds the lock on the worker directory
-    (try-cleanup-worker conf id))
+    (try-cleanup-worker conf id)
+    (if (conf STORM-CGROUP-ENABLE)
+      (.shutDownWorker (:cgroup-manager supervisor) id false)))
```

Comment:
I would move check and call for `shutdownWorker` under `try-cleanup-worker` method itself. Also, we can check if `(not-nil? (:cgroup-manger supervisor) )`.